tiledb/common/src/filter/webp.rs

```diff
@@ -1,11 +1,19 @@
+use thiserror::Error;
+
+#[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
-use util::option::OptionSubset;
+#[cfg(feature = "option-subset")]
+use tiledb_utils::option::OptionSubset;
 
-use crate::Result as TileDBResult;
+#[derive(Clone, Debug, Error)]
+pub enum WebPFilterError {
+    #[error("Invalid discriminant for {}: {0}", std::any::type_name::<WebPFilterInputFormat>())]
+    InvalidDiscriminant(u64),
+}
 
-#[derive(
-    Copy, Clone, Debug, Deserialize, Eq, OptionSubset, PartialEq, Serialize,
-)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "option-subset", derive(OptionSubset))]
```

Comment:
Another feature that I'm not 100% sold on. Easy to change down the line, so not requesting a change. Just seems a bit "feature happy" is all.